### ISSUE TITLE
@SHF with @ASN regridding yields inconsistent behavior at outer edge point

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#1573]
**Reported by andrew wittenberg on 11 Jun 2015 17:20 UTC**
Define some variables on a staggered grid:

```
 	NOAA/PMEL TMAP
 	FERRET v6.95 (beta)
 	Linux 2.6.32-504.16.2.el6.x86_64 64-bit - 06/09/15
 	11-Jun-15 13:17     

yes? def ax/edge/z=0:40:10 zax_t
yes? def ax/edge/z=5:45:10 zax_w
yes? let zw = z[let zdiff = zw - missing(zw[z=@shf:-1](gz=zax_w]
yes?),0)
yes? let zdiff_t = zdiff[zdiff_t ought to be well-defined everywhere.  But if we try to evaluate at k=1, we get an error:

```
yes? list/k=1 zdiff_t
 **ERROR: illegal limits: ZW does not exist at K=0
          Axis extremes are K=1:4
```

But listing the full field *does* display a value at k=1:

```
yes? list zdiff_t
             VARIABLE : ZDIFF[GZ=ZAX_T@ASN](gz=zax_t@asn]
```

Now)
             SUBSET   : 4 points (Z)
 5    / 1:  10.00
 15   / 2:  10.00
 25   / 3:  10.00
 35   / 4:  10.00
```

And now something has changed.  Our previous command which failed, now produces the expected result!

```
yes? list/k=1 zdiff_t
             VARIABLE : ZDIFF[GZ=ZAX_T@ASN]
             Z        : 5
          10.00
```

What's going on?

See also #1462, in case it is related.

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/2301

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 11 Jun 2015 22:48 UTC**
When getting ready to compute the @SHF transformation, the index offsets are applied. The new index range may take us beyond the ends of the axis, e.g. k=0.  If the upper index value is still within the axis range, this gets handled.  If both the upper and lower indices are beyond the axis, so lo=hi=0, a call to analyze the context throws an error.

I'm looking at where we might test for this. There are (1992) comments in the code about this known issue.

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 12 Jun 2015 19:11 UTC**
This is fixed. These are cases where the requested region is at the end of an axis, and the index offset for the transformation takes the high subscript below the axis start or the low subscript above the axis end.

**Comment by karlmsmith** [comment]
**Modified by ansley.b.manke on 12 Jun 2015 19:15 UTC**

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 14 Jun 2015 00:23 UTC**
This fix also appears to have resolved #2300 and #1462.
